lib/plugins/ftp/index.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const log = require('intel').getLogger('sitespeedio.plugin.ftp');
+const PromiseFtp = require('promise-ftp');
+const merge = require('lodash.merge');
+const throwIfMissing = require('../../support/util').throwIfMissing;
+const fs = require('fs-extra');
+const defaultConfig = {
+  host: 'localhost',
+  port: 21,
+  user: 'user',
+  password: 'password'
+};
+function createFTPClient(ftpOptions) {
+  const ftpOptions = merge({}, defaultConfig, options.ftp);
+  var ftp = new PromiseFtp();
+  return ftp.connect(ftpOptions)
+}
+
+module.exports = {
+  open(context, options) {
+    this.ftpOptions = options.ftp;
+    this.make = context.messageMaker('ftp').make;
+
+    throwIfMissing(this.ftpOptions, ['host'], 'ftp');
+    throwIfMissing(this.ftpOptions, ['user', 'password'], 'ftp');
+    this.storageManager = context.storageManager;
+  },
+
+  processMessage(message, queue) {
+    const make = this.make;
+    if (message.type === 'html.finished') {
+      const ftpOptions = merge({}, defaultConfig, this.ftpOptions);
+      const baseDir = this.storageManager.getBaseDir();
+      const storagePathPrefix = this.storageManager.getStoragePrefix();
+      var client = new PromiseFtp();
+
+      return new Promise((resolve, reject) => {
+        log.info(`FTP uploading output (${baseDir}) to FTP ${ftpOptions.user}@${ftpOptions.host}, this can take a while ...`);
+        var dirlist = [];
+        var filelist = [];
+
+        function walkSync(dir) {
+          var path = path || require('path');
```

Comment:
We usually do all requires at the top of the file to make it easy to see what's used, so if you can move it, thanks!